### ISSUE TITLE
SQL Error when updating athlete info

### ISSUE THREAD

**Issue by privong** [pytrainer/pytrainer#151]
I am using pytrainer version 1.12.1 and attempting to add athlete information results in a crash due to a string being passed in the INSERT statement when it should be float:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/pytrainer/gui/windowmain.py", line 1640, in on_newrecord_clicked
    self.on_athleteTreeView_edit( None, None)
  File "/usr/lib/python2.7/site-packages/pytrainer/gui/windowmain.py", line 2087, in on_athleteTreeView_edit
    self.on_athleteSave(data)
  File "/usr/lib/python2.7/site-packages/pytrainer/gui/windowmain.py", line 2129, in on_athleteSave
    self.parent.athlete.insert_athlete_stats(date, weight, bodyfat, restinghr, maxhr)
  File "/usr/lib/python2.7/site-packages/pytrainer/athlete.py", line 141, in insert_athlete_stats
    self.pytrainer_main.ddbb.session.commit()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 943, in commit
    self.transaction.commit()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 467, in commit
    self._prepare_impl()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 447, in _prepare_impl
    self.session.flush()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2254, in flush
    self._flush(objects)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2380, in _flush
    transaction.rollback(_capture_exception=True)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 66, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/session.py", line 2344, in _flush
    flush_context.execute()
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 391, in execute
    rec.execute(self)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 556, in execute
    uow
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 181, in save_obj
    mapper, table, insert)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 866, in _emit_insert_statements
    execute(statement, params)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 948, in execute
    return meth(self, multiparams, params)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1132, in _execute_context
    None, None)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
    exc_info
  File "/usr/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1127, in _execute_context
    context = constructor(dialect, self, conn, *args)
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 667, in _init_compiled
    param.append(processors[key](compiled_params[key]))
sqlalchemy.exc.StatementError: (exceptions.ValueError) could not convert string to float:  [SQL: u'INSERT INTO athletestats (bodyfat, date, maxhr, restinghr, weight) VALUES (?, ?, ?, ?, ?)'] [parameters: [{'date': datetime.date(2018, 5, 14), 'weight': '78.9', 'restinghr
'', 'bodyfat': '', 'maxhr': ''}]]
```

I receive a similar error for the body fat, resting heart rate, and maximum heart rate fields if I try to submit information in those as well.

**Comment by viiru-** [comment]
Certain parts of this issue have been fixed in version 1.12.2, and the rest is fixed in master.

**Comment by privong** [comment]
Great, thanks.
